src/command_modules/azure-cli-servicebus/azure/cli/command_modules/servicebus/custom.py

```diff
@@ -458,3 +472,48 @@ def return_valid_duration_create(update_value):
                 return None
     else:
         return None
+
+
+# NetwrokRuleSet Region
+def cli_networkrule_createupdate(client, resource_group_name, namespace_name, subnet=None, ip_mask=None, ignore_missing_vnet_service_endpoint=False, action='Allow'):
+    from azure.mgmt.servicebus.models import NWRuleSetVirtualNetworkRules, Subnet, NWRuleSetIpRules
+    netwrokruleset = client.get_network_rule_set(resource_group_name, namespace_name)
+
+    if netwrokruleset.virtual_network_rules is None:
+        netwrokruleset.virtual_network_rules = [NWRuleSetVirtualNetworkRules]
+
+    if netwrokruleset.ip_rules is None:
+        netwrokruleset.ip_rules = [NWRuleSetIpRules]
+
+    if subnet:
+        netwrokruleset.virtual_network_rules.append(NWRuleSetVirtualNetworkRules(subnet=Subnet(id=subnet),
+                                                                                 ignore_missing_vnet_service_endpoint=ignore_missing_vnet_service_endpoint))
+
+    if ip_mask:
+        netwrokruleset.ip_rules.append(NWRuleSetIpRules(ip_mask=ip_mask, action=action))
+
+    return client.create_or_update_network_rule_set(resource_group_name, namespace_name, netwrokruleset)
+
+
+def cli_networkrule_delete(client, resource_group_name, namespace_name, subnet=None, ip_mask=None):
+    from azure.mgmt.servicebus.models import NWRuleSetVirtualNetworkRules, NWRuleSetIpRules
+    netwrokruleset = client.get_network_rule_set(resource_group_name, namespace_name)
+
+    if subnet:
+        virtualnetworkrule = NWRuleSetVirtualNetworkRules()
+        virtualnetworkrule.subnet = subnet
+
+        for vnetruletodelete in netwrokruleset.virtual_network_rules:
+            if vnetruletodelete.subnet.id == subnet:
+                virtualnetworkrule.ignore_missing_vnet_service_endpoint = vnetruletodelete.ignore_missing_vnet_service_endpoint
+                netwrokruleset.virtual_network_rules.remove(vnetruletodelete)
```

Comment:
You should be able to `break` after a match.

Reply:
resolved